subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/ContinuousBuildActionExecutor.java

```diff
@@ -163,6 +179,9 @@ private void waitForDeployments(BuildAction action, BuildRequestMetaData request
                 if (buildInputs.isEmpty()) {
                     logger.println().withStyle(StyledTextOutput.Style.Failure).println("Exiting continuous build as no executed tasks declared file system inputs.");
                     return lastResult;
+                } else if (!continuousBuildTriggerHandler.hasBeenTriggered() && !fileSystemWatchingInformation.isWatchingAnyLocations()) {
+                    logger.println().withStyle(StyledTextOutput.Style.Failure).println("Exiting continuous build as Gradle does not watch any file system locations.");
```

Comment:
Is that enough information for a regular user, so the user will know the actual reason? Should we add some documentation link here, with why this is an issue? 
I as a user would ask myself, why Gradle does not watch any file system, I just wanted to run a continuous build and now I got an error.

Reply:
I think having a link to documentation would be a cool thing. Will you do it in this PR or a different one?